src/runner.rs

```diff
@@ -1419,37 +1311,47 @@ impl ValueStack {
 
 	#[inline]
 	fn top(&self) -> &RuntimeValueInternal {
-		self.pick(1)
-	}
-
-	fn pick(&self, depth: usize) -> &RuntimeValueInternal {
-		&self.buf[self.sp - depth]
+		self.0.top().expect("pre-validated")
```

Comment:
Please make this error message better, something like: "The stack is empty, this should never be possible due to the validation step".